pennylane/templates/subroutines/semi_adder.py

```diff
@@ -0,0 +1,285 @@
+            work_wires (Sequence[int]): the auxiliary wires to use for the addition. Exactly ``len(y_wires) - 1`` work
+                wires should be provided.
+
+        Returns:
+            list[.Operator]: Decomposition of the operator
+        """
+        op_list = []
+
+        # revert wires to follow PennyLane convention
+
+        x_wires_pl = x_wires[::-1][: len(y_wires)]
+        y_wires_pl = y_wires[::-1]
+        work_wires_pl = work_wires[::-1]
+        op_list.append(qml.TemporaryAND([x_wires_pl[0], y_wires_pl[0], work_wires_pl[0]]))
+
+        for i in range(1, len(y_wires_pl) - 1):
+            if i < len(x_wires_pl):
+                op_list += _left_operator(
+                    work_wires_pl[i - 1], x_wires_pl[i], y_wires_pl[i], work_wires_pl[i]
+                )
+            else:
+                # If the number of qubits in |x> is smaller than |y>, we can conceptually complete the |x> state
+                # with |0> on the left making sure they are of the same size. Assuming this, we can simplify the left operator.
+                op_list.append(qml.CNOT([work_wires_pl[i - 1], y_wires_pl[i]]))
+                op_list.append(
+                    qml.TemporaryAND([work_wires_pl[i - 1], y_wires_pl[i], work_wires_pl[i]])
+                )
+                op_list.append(qml.CNOT([work_wires_pl[i - 1], work_wires_pl[i]]))
+
+        op_list.append(qml.CNOT([work_wires_pl[-1], y_wires_pl[-1]]))
+
+        if len(x_wires_pl) >= len(y_wires_pl):
+            op_list.append(qml.CNOT([x_wires_pl[-1], y_wires_pl[-1]]))
+
+        for i in range(len(y_wires_pl) - 2, 0, -1):
+            if i < len(x_wires_pl):
+                op_list += _right_operator(
+                    work_wires_pl[i - 1], x_wires_pl[i], y_wires_pl[i], work_wires_pl[i]
+                )
+            else:
+                # If the number of qubits in |x> is smaller than |y>, we can conceptually complete the |x> state
+                # with |0> on the left making sure they are of the same size. Assuming this, we can simplify the right operator.
+                op_list.append(qml.CNOT([work_wires_pl[i - 1], work_wires_pl[i]]))
+                op_list.append(
+                    qml.adjoint(
+                        qml.TemporaryAND([work_wires_pl[i - 1], y_wires_pl[i], work_wires_pl[i]])
+                    )
+                )
+
+        op_list.append(
+            qml.adjoint(qml.TemporaryAND([x_wires_pl[0], y_wires_pl[0], work_wires_pl[0]]))
+        )
+
+        op_list.append(qml.CNOT([x_wires_pl[0], y_wires_pl[0]]))
+
+        return op_list
+
+
+def _semiadder_resources(num_y_wires):
+    # Resources extracted from `arXiv:1709.06648 <https://arxiv.org/abs/1709.06648>`_.
+    # In the case where len(x_wires) < len(y_wires), this is an upper bound.
+    return {
+        qml.TemporaryAND: num_y_wires - 1,
+        qml.decomposition.adjoint_resource_rep(qml.TemporaryAND, {}): num_y_wires - 1,
```

Comment:
Why did you have to use the resource_rep here? does `qml.Adjoint(qml.TemporaryAND)` not work?

Reply:
Sadly it doesn't work in that way

Reply:
Got it 👍🏼